collect_app/src/androidTest/java/org/odk/collect/android/tasks/FormLoaderTaskTest.java

```diff
@@ -0,0 +1,38 @@
+package org.odk.collect.android.tasks;
+
+import android.Manifest;
+
+import androidx.test.rule.GrantPermissionRule;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.support.CopyFormRule;
+import org.odk.collect.android.support.ResetStateRule;
+
+import java.io.File;
+import java.util.Arrays;
+
+public class FormLoaderTaskTest {
+    private static final String EXTERNAL_CSV_FORM = "external_csv_form.xml";
+
+    @Rule
+    public RuleChain copyFormChain = RuleChain
+            .outerRule(GrantPermissionRule.grant(
+                    Manifest.permission.READ_EXTERNAL_STORAGE,
+                    Manifest.permission.WRITE_EXTERNAL_STORAGE
+            ))
+            .around(new ResetStateRule())
+            .around(new CopyFormRule(EXTERNAL_CSV_FORM,
+                    Arrays.asList("external_csv_cities.csv", "external_csv_countries.csv", "external_csv_neighbourhoods.csv")));
+
+    @Test
+    public void loadFormWithSecondaryCSV() throws Exception {
+        final String formPath = Collect.FORMS_PATH + File.separator + EXTERNAL_CSV_FORM;
+        FormLoaderTask formLoaderTask = new FormLoaderTask(formPath, null, null);
+        FormLoaderTask.FECWrapper wrapper = formLoaderTask.execute(formPath).get();
+        Assert.assertNotNull(wrapper);
```

Comment:
We've generally written these kinds of Espresso form loading tests to display the form and assert something on the form. See `ExternalSecondaryInstancesTest` for an example with an XML external secondary instance. That makes it more end-to-end for that form definition.

I think you were (rightfully) looking for a way to make the test more targeted, right? The wrapper being null is a proxy for the task failing?

I wonder whether since we don't have any tests for secondary CSV instances and instrumented tests are expensive anyway we should go all the way and verify that the form entry can be done correctly.

Reply:
Yes, I was trying to make a targeted test but I didn't want to make big refactoring changes to FormLoaderTask to do so. With tests like this in place we can proceed to make the refactoring changes needed to make the test even smaller. I'd rather go that route than make the test bigger to verify form entry. But I understand how hijacking an intermediate loading stage for this testing purpose is a bit deviant :)

Reply:
> a bit deviant :)

Was trying not to say it that way but yes... ;)